app/code/core/Mage/Api/Model/Resource/User.php

```diff
@@ -335,14 +335,14 @@ public function _getRoles(Mage_Core_Model_Abstract $user)
     /**
      * Add Role
      *
-     * @param Mage_Core_Model_Abstract $user
+     * @param Mage_Core_Model_Abstract|Mage_Api_Model_User $user
      * @return $this
      */
     public function add(Mage_Core_Model_Abstract $user)
     {
         $adapter = $this->_getWriteAdapter();
         $aRoles  = $this->hasAssigned2Role($user);
-        if (sizeof($aRoles) > 0) {
+        if (is_array($aRoles) && count($aRoles) > 0) {
```

Comment:
code change

Reply:
Cause by wrong DOC. It is an array in any case ...